helix-rest/src/main/java/org/apache/helix/rest/server/resources/metadatastore/MetadataStoreDirectoryAccessor.java

```diff
@@ -153,9 +161,13 @@ public Response getShardingKeys(@QueryParam("realm") String realm) {
   public Response addShardingKey(@PathParam("realm") String realm,
       @PathParam("sharding-key") String shardingKey) {
     try {
-      _metadataStoreDirectory.addShardingKey(_namespace, realm, shardingKey);
+      if (!_metadataStoreDirectory.addShardingKey(_namespace, realm, shardingKey)) {
+        return serverError();
```

Comment:
Can we have a message in the server error message why it fails or what the client should do? Server error is too plain.

Reply:
There are numerous different reasons why this could happen, and one single returned message is not going to cover all cases; saying it's a server error is sufficient to express that it's a server side situation. 
On the other hand, whenever `false` is returned here, there are server logs that can help us identify the cause.

Reply:
That is for server side. Think about client side: what is response for this? If you are a user, what would you like to know?

Reply:
Here are the situations where this condition can be invoked:
1. When the request forwarded to the leader instance returns a non-ok;
2. When the request forwarded to the leader raises an IOException;
3. When writing to ZooKeeper fails. 
All of them happen on the layer of `ZkRoutingDataWriter`. I'll need some advice on how to propagate the correct information back to the REST layer - I do not see an easy way to do this. 
At the same time, I believe it's unnecessary to let clients know the exact situation among these 3 situations, because these situations are unrelated to client input and there's nothing clients can do. All these situations are strictly internal errors and are out of the hands of clients.

Reply:
I understand the possible situations and there is no need to tell the client the real reason (and it is impossible here because the api just returns true or false). I meant, include a message suggesting what to do like “server error , please contact admin” is helpful than plain response. Imagine, what would you get if you do curl this endpoint but server error? 
Anyway, I have an issue Created to improve the response. You don’t have to do it here. Focus on the main task of the pr instead.